### PR TITLE
Task 68/simultaneous key presses

### PR DESCRIPTION
**Pull request by MrDmitry**

### CODE REVIEW

File: UnicornEngine/system/source/Manager.cpp [hunk 1]
```diff
@@ -294,5 +304,19 @@ void Manager::PollGamepads()
     }
 }
 
+void Manager::PollWindows()
+{
+    for (auto const& cit : m_windows)
+    {
+        if (cit.second->IsFocused())
```
Comment (MrDmitry) [inline UnicornEngine/system/source/Manager.cpp]: Not sure if maybe all windows shall be polled for events. Should not cause much overhead I dont think

File: UnicornEngine/system/include/unicorn/system/Window.hpp [hunk 2]
```diff
@@ -325,16 +308,18 @@ class Window
...
-     *  -# key indicator
-     *  -# raw scancode value
-     *  -# key action type
-     *  -# modifiers mask
-     */
-    wink::signal< wink::slot<void(Window*, input::Key, uint32_t, input::Action, input::Modifier::Mask)> > Keyboard;
+    /** @brief  Keyboard event information */
+    struct KeyboardEvent
+    {
+        Window * const pWindow;
```
Comment (krupitskas) [inline UnicornEngine/system/include/unicorn/system/Window.hpp]: Pointer alignment

File: UnicornEngine/system/source/profiler/KeyProfiler.cpp [hunk 3]
```diff
@@ -41,8 +40,14 @@ void KeyProfiler::OnWindowCreated(Window* pWindow)
     pWindow->Unicode.connect(this, &KeyProfiler::OnWindowUnicode);
 }
 
-void KeyProfiler::OnWindowKeyboard(Window* pWindow, input::Key key, uint32_t scancode, input::Action action, input::Modifier::Mask modifiers)
+void KeyProfiler::OnWindowKeyboard(Window::KeyboardEvent const& keyboardEvent)
 {
+    Window * const& pWindow = keyboardEvent.pWindow;
```
Comment (krupitskas) [inline UnicornEngine/system/source/profiler/KeyProfiler.cpp]: pointer alignment?

File: UnicornEngine/system/include/unicorn/system/Window.hpp [hunk 4]
```diff
@@ -291,15 +272,17 @@ class Window
...
-     *  -# window pointer
-     *  -# mouse button input
-     *  -# button action type
-     *  -# modifiers mask
-     */
-    wink::signal< wink::slot<void(Window*, input::MouseButton, input::Action, input::Modifier::Mask)> > MouseButton;
+    /** @brief  Mouse button event information */
+    struct MouseButtonEvent
+    {
+        Window* const pWindow;
```
Comment (ilia-glushchenko) [inline UnicornEngine/system/include/unicorn/system/Window.hpp:278]: Are you not going to store these events?

Comment (MrDmitry) [reply UnicornEngine/system/include/unicorn/system/Window.hpp:278]: They are stored inside `wink::event_queue MouseButton` until `MouseButton.emit()`

**Comment by krupitskas** [comment]
Approve

[Approved with PullApprove](https://pullapprove.com/Godlike/Unicorn/pull-request/73/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Godlike/Unicorn) [Approved with PullApprove](https://pullapprove.com/Godlike/Unicorn/pull-request/73/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Godlike/Unicorn)

**Comment by ilia-glushchenko** [comment]
Approve

[Approved with PullApprove](https://pullapprove.com/Godlike/Unicorn/pull-request/73/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Godlike/Unicorn) [Approved with PullApprove](https://pullapprove.com/Godlike/Unicorn/pull-request/73/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Godlike/Unicorn)

**Pull request merged**